java/src/main/native/include/jni_utils.hpp

```diff
@@ -790,19 +760,43 @@ inline void jni_cuda_check(JNIEnv *const env, cudaError_t cuda_status) {
     JNI_THROW_NEW(env, class_name, message, ret_val)                                               \
   }
 
-#define JNI_CUDA_TRY(env, ret_val, call)                                                           \
-  {                                                                                                \
+// Throw a new exception only if one is not pending then always return with the specified value
+#define JNI_CHECK_THROW_NEW_CUDA_ERROR(env, class_name, e, is_halt, ret_val)                       \
+  do {                                                                                             \
+    if (env->ExceptionOccurred()) {                                                                \
+      return ret_val;                                                                              \
+    }                                                                                              \
+    std::string n_msg = e.what() == nullptr ? "" : e.what();                                       \
+    jstring j_msg = env->NewStringUTF(n_msg.c_str());                                              \
+    jint e_code = static_cast<jint>(e.error_code());                                               \
+    jclass ex_class = env->FindClass(class_name);                                                  \
+    jmethodID ctor_id = env->GetMethodID(ex_class, "<init>", "(Ljava/lang/String;I)V");            \
+    if (ctor_id == NULL) {                                                                         \
+      return ret_val;                                                                              \
+    }                                                                                              \
+    jobject cuda_error = env->NewObject(ex_class, ctor_id, j_msg, e_code);                         \
+    env->Throw((jthrowable)cuda_error);                                                            \
+    if (is_halt) {                                                                                 \
+      cudf::jni::throw_jni_exception("CUDA ERROR");                                                \
+    }                                                                                              \
+    return ret_val;                                                                                \
+  } while (0)
+
+#define JNI_CUDA_TRY(ret_val, call)                                                                \
```

Comment:
This seems like we've re-invented `CUDF_CUDA_TRY`, and I think we should simply use that directly instead.

Reply:
Removed.